### PR TITLE
use colcon instead of catkin_make_isolated

### PR DESCRIPTION
**Pull request by dirk-thomas**
For now the patch avoids any kind of renaming to keep it as close to master as possible.

### CODE REVIEW

File: scripts/devel/catkin_make_isolated_and_test.py [hunk 1]
```diff
@@ -61,6 +61,10 @@ def main(argv=sys.argv[1:]):
     if args.parent_result_space:
         parent_result_spaces = args.parent_result_space
 
+    # TODO try to remove melodic from this list
+    use_merge_install = args.rosdistro_name in (
+        'indigo', 'jade', 'kinetic', 'lunar', 'melodic')
```
[the remark this answers is not in the document]
Comment (dirk-thomas) [reply scripts/devel/catkin_make_isolated_and_test.py]: We could but I don't see a need for it. If anyone would still run a Jade buildfarm keeping the entry will ensure to not change the behavior.

**Comment by dirk-thomas** [comment]
The current state of the patch seems solid enough to build the ROS 1 devel jobs minus the differences of cmi in terms of accessing resources from the source space of other packages in the repo (see https://github.com/ros-infrastructure/ros_buildfarm_config/pull/117#issuecomment-395271508).

I will rewrite the changes in a small set of easier to review commits...

**Comment by mikaelarguedas** [comment]
@dirk-thomas Running some prerelease jobs I faced a few things that impacted user experience:
- Removal of `echo "'catkin_test_results' not found on the PATH. Please install catkin and source the environment to output the test result summary."` https://github.com/ros-infrastructure/ros_buildfarm/pull/548/files#diff-fcd271523769c381d8669fe60bcba665L13. Now if colcon is not sourced / on the path it just prints `colcon command not found`
- No indicator of the progress of the builds, is the status line expected to be printed ?
(with catkin_make isolated each package starting to build printed "package X of Y")

**Comment by dirk-thomas** [comment]
> if colcon is not sourced / on the path it just prints `colcon` command not found

This has been address in #585 printing a similar error message as for `catkin_make_isolated`.

> No indicator of the progress of the builds, is the status line expected to be printed ?

The status line is not expected to be printed since it is only enabled when using an interactive shell. But you should still see the progress of the build (start / finish messages per package as well as the output of the build). Please try again with #585 and comment in which exact scenario you don't see any progress.

**Comment by dirk-thomas** [comment]
This PR has been superseded by #585 and will be closed once the existing jobs have been transitioned to the new PR (or to master after it has been merged).

**Comment by dirk-thomas** [comment]
The Bouncy devel jobs have been reconfigured to use the `master` branch.

**Pull request closed without merging**